### ISSUE TITLE
Makeinstall.bat wont show any usb

### ISSUE THREAD

**Issue by namide1** [corpnewt/gibMacOS#179]
I read almost every fix for this issue, commands like wmic logicaldisk get deviceid,filesystem,volumename,size,drivetype /format:csv not work, clicking d on keyboard and turning on gaining no solution, help.
My spec:
Motherboard: Msi h110m-pro-vd-plus
Ram: 8gb single channel goodram white
Processor: i5 7400
Graphics card: Msi Gtx 1050 ti oc 4g
drives: normal, casual

**Comment by namide1** [comment]
![image](https://user-images.githubusercontent.com/73107146/96430351-55951400-1202-11eb-8de0-8fe6bd648dff.png)

**Comment by namide1** [comment]
can i put number from diskpart (list disk)?

**Comment by love-master** [comment]
I have the same problem, I do not know how to get the drive number.

**Comment by gamecss** [comment]
please press D and Enter

**Comment by gamecss** [comment]
or delete your usb's part

**Comment by stavitian** [comment]
i did both without success , windows 11 Btw

**Comment by NorthernFlow** [comment]
same problem. windows 11. It seems that this program doesnt work on w11...

**Comment by RavHilton** [comment]
Same issue here - no drives are shown (Win 11, latest public build, latest public updates, Python 3.11.1, x64) and the addition of disk_type "3" as suggested by https://github.com/wilkergrcv makes no difference at all. Yes, already running elevated as admin...

Even after using option "D", all I see is as follows:

#######################################################

Potential Removable Media
#######################################################

!WARNING! This list includes ALL disk types.
!WARNING! Be ABSOLUTELY sure before selecting
!WARNING! a disk!

Q. Quit

Usage:

**Comment by KuusamoMart** [comment]
Another Windows 11 user with the same problem.

**Comment by Masamune3210** [comment]
Ok, think I figured it out. The script uses WMIC to get the disks attached to the system. WMIC because a by default disabled optional feature in Windows 11 Build 22572. Someone who is having this issue, open up the optional features dialog and see if WMIC or Windows Management Instrumentation is disabled. If it is, enable it, reboot, and then try the script again.

**Comment by KuusamoMart** [comment]
I'm using Windows 11 build 22621.1702; the optional feature dialogue does NOT feature WMIC, despite it being an optional feature on build 22572.
The suggested option from MS is to use powershell commands, but I've looked at the documentation and couldn't even begin to work out how to use the powershell command structure in place of WMIC.

**Comment by KuusamoMart** [comment]
So it turns out you can add via the Powershell DISM command. Did that, now have WMIC, but the disks still don't show!

**Comment by KuusamoMart** [comment]
So, with WMIC installed in Window 11, the drives are still not shown. I found an article that indicates this is a known issue with /format:csv on a non US English Windows installation. In my case, csv.xsl is located in c:\windows\system32\wbem\fi-FI\, however WMIC cannot locate it unless I put the full path name in front of csv. Adding the above folder to PATH doesn't work (got to love how inconsistent Windows is!)

**Comment by KuusamoMart** [comment]
This is the line in diskwin.py that's causing issues:
disks = self.r.run({"args":[self.wmic, "diskdrive", "get", "deviceid,model,index,size,partitions", "/format:csv"]})[0]
If I run wmic diskdrive get format:csv on the command line, it fails. If I run wmic diskdrive get /format:"c:\windows\system32\wbem\fi-FI\csv" it works and outputs the csv string. What I am unsure of is how to put this path within the arguments of the above line of Python code.

**Comment by Masamune3210** [comment]
It's likely something not getting passed correctly, embedding paths into a function call like that is pushed against pretty hard for that reason exactly. Honestly, it would probably be better in the end to just rewrite the disk detection to call into something else like DISM. Unless I'm misremembering, all it uses WMIC for is detecting disks, types, sizes, and getting a handle to the physical disk that is selected. Could be wrong though, GitHub mobile sucks for this kind of introspection

**Comment by KuusamoMart** [comment]
Agreed, forcing a path like that is horrible, but I have no idea what else to try!
You are correct, WMIC is being used to find the drive details.

**Comment by corpnewt** [comment]
Do the following powershell commands return information properly on Windows 11 machines?:
```
Get-WmiObject -Class Win32_DiskDrive
Get-WmiObject -Class Win32_LogicalDisk
Get-WmiObject -Class Win32_LogicalDiskToPartition
```
If so - I could likely check for `where wmic` first - and use powershell as a fallback as needed to parse the disks.

-CorpNewt

**Comment by KuusamoMart** [comment]
Yes, all 3 return sensible results. Tested on Windows 11 Pro

**Comment by corpnewt** [comment]
Alright - I'll be at work for much of today, but if I have time I'll see what I can whip up.

-CorpNewt

**Comment by Masamune3210** [comment]
Mind you, I think where is cmd only, on PowerShell it's Get-Location

**Comment by corpnewt** [comment]
I would be running `where` directly via Python's subprocess module, so that shouldn't impact anything. Thank you for the information though. I've used PowerShell a handful of times, and it always seems to behave differently than it should lol.

-CorpNewt

**Comment by corpnewt** [comment]
I think I've got something tangible set up - can you test as of [this commit](https://github.com/corpnewt/gibMacOS/commit/054fa19306676d2475431730c0492833824e4804)?

The `diskwin.py` rework should check for the existence of `wmic` first - and if located, will use that.  If it does not find `wmic`, it will attempt to gather/parse info from `powershell` instead.  It works on my Windows 10 machine with the `wmic` detection, and forcing it to use `powershell`.  Please let me know if that does, or doesn't fix this issue.

-CorpNewt

**Comment by Masamune3210** [comment]
Sorry, its 9:41 PM here, if someone doesn't get to it before I get off of work tomorrow, I will test and let you know

**Comment by KuusamoMart** [comment]
I have tested it and it's looking good. I did have to remove the WCIM feature that I had enabled in Windows 11 first, as it was found during my initial test and had the same issue as previously, Once it was removed, it behaved as intended.
Thanks for the quick fix!

**Comment by corpnewt** [comment]
Hmm - I should be able to have it fall back to PowerShell if WMIC doesn't locate any disks (or the commands fail for whatever reason).  I'll see if I can implement that before work today.

-CorpNewt

**Comment by youssef-mostafa1534** [comment]
at this point there are multiple threads about this issue, it has not been solved yet for everyone, and what we know about it is that it works on windows 10 and not 11

**Comment by youssef-mostafa1534** [comment]
do older versions of gibmacos work on windows 11?, because it doesn't matter to me if it's an out dated version i'd use it as long as it works
